### PR TITLE
Create get models statuses implementations

### PR DESCRIPTION
**Pull request by michalkulakowski**

### CODE REVIEW

File: src/model_service.cpp [hunk 1]
```diff
@@ -132,6 +132,41 @@ Status GetModelStatusImpl::getModelStatus(
...
+        output_tmp += ("{\n\"" + model_version.first + "\" : \n");
+        std::string response_str;
+        auto status = GetModelStatusImpl::serializeResponse2Json(&model_version.second, &response_str);
+        if(status != StatusCode::OK)
+        {
+            return status;
+        }
+        output_tmp += (response_str + "}") ;
+    }
+    *output = output_tmp;
```
Comment (mzegla) [inline src/model_service.cpp:165]: Can we use reference here for the `output`? If not you need to make sure `output` is a valid pointer.

File: src/model_service.hpp [hunk 2]
```diff
@@ -52,4 +52,10 @@ class GetModelStatusImpl {
     static Status serializeResponse2Json(const tensorflow::serving::GetModelStatusResponse* response, std::string* output);
 };
 
+class GetModelsImpl {
```
Comment (mzegla) [inline src/model_service.hpp:55]: Do we actually need this class? Maybe those methods could go under `GetModelStatusImpl`?

File: src/test/model_service_test.cpp [hunk 3]
```diff
@@ -313,3 +313,225 @@ TEST(RestModelStatus, serialize2Json) {
...
+    GetModelStatusImpl::getAllModelsStatuses(modelsStatusesAfterReload, manager);
+    
+    ASSERT_EQ(modelsStatusesAfterReload.size(), 1);
+    EXPECT_EQ(modelsStatusesAfterReload.begin()->second.model_version_status_size(), 1);
+    EXPECT_EQ(modelsStatusesAfterReload.begin()->second.model_version_status().begin()->state(), tensorflow::serving::ModelVersionStatus_State_AVAILABLE);
+    EXPECT_EQ(modelsStatusesAfterReload.begin()->second.model_version_status().begin()->version(), 1);
+    EXPECT_EQ(modelsStatusesAfterReload.begin()->second.model_version_status().begin()->has_status(), true);
+    EXPECT_EQ(modelsStatusesAfterReload.begin()->second.model_version_status().begin()->status().error_code(), tensorflow::error::OK);
+    EXPECT_EQ(modelsStatusesAfterReload.begin()->second.model_version_status().begin()->status().error_message(), "OK");
+    EXPECT_EQ(modelsStatusesAfterReload.begin()->second.model_version_status_size(), 1);
```
Comment (mzegla) [inline src/test/model_service_test.cpp:343]: Duplicate of line 337.

File: src/test/model_service_test.cpp [hunk 4]
```diff
@@ -313,3 +313,225 @@ TEST(RestModelStatus, serialize2Json) {
...
+        EXPECT_EQ(response.state(), tensorflow::serving::ModelVersionStatus_State_AVAILABLE);
+        EXPECT_EQ(response.version(), i + 1);
+        EXPECT_EQ(response.has_status(), true);
+        EXPECT_EQ(response.status().error_code(), tensorflow::error::OK);
+        EXPECT_EQ(response.status().error_message(), "OK");
+    }
+}
+
+TEST(ModelService, serializeModelsStatuses2Json_with_one_response) {
+    const char* expected_json = R"({
```
Comment (mzegla) [inline src/test/model_service_test.cpp:415]: Can we have camel case here?

File: src/test/model_service_test.cpp [hunk 5]
```diff
@@ -313,3 +313,225 @@ TEST(RestModelStatus, serialize2Json) {
...
+}
+})";
+    tensorflow::serving::GetModelStatusResponse response;
+    model_version_t requested_version = 2;
+    const std::string& model_name = "dummy";
+    ModelVersionStatus status = ModelVersionStatus(model_name, requested_version, ModelVersionState::START);
+    addStatusToResponse(&response, requested_version, status);
+    std::map<std::string, tensorflow::serving::GetModelStatusResponse> modelsStatuses;
+    modelsStatuses.insert(std::pair<std::string, tensorflow::serving::GetModelStatusResponse>("dummy", response));
+    std::string json_output;
```
Comment (mzegla) [inline src/test/model_service_test.cpp:437]: Can we have camel case here?

File: src/test/model_service_test.cpp [hunk 6]
```diff
@@ -313,3 +313,225 @@ TEST(RestModelStatus, serialize2Json) {
...
+    Status error_status = GetModelStatusImpl::serializeModelsStatuses2Json(modelsStatuses, json_output);
```
Comment (mzegla) [inline src/test/model_service_test.cpp:438]: Can we have camel case here?

**Review by mzegla — APPROVED** [review]

File: src/model_service.cpp [hunk 7]
```diff
@@ -132,6 +132,44 @@ Status GetModelStatusImpl::getModelStatus(
...
+
+    return StatusCode::OK;
+}
+
+Status GetModelStatusImpl::serializeModelsStatuses2Json(const std::map<std::string, tensorflow::serving::GetModelStatusResponse>& modelsStatuses, std::string& output){
+    std::string outputTmp;
+    for(auto modelStatus = modelsStatuses.begin(); modelStatus != modelsStatuses.end(); modelStatus++)
+    {
+        outputTmp += ("{\n\"" + modelStatus->first + "\" : \n");
+        std::string response_str;
```
Comment (mzegla) [inline src/model_service.cpp:157]: Can we stick to camelCase?

**Review by mzegla — APPROVED** [review]

File: src/model_service.cpp [hunk 8]
```diff
@@ -132,6 +132,44 @@ Status GetModelStatusImpl::getModelStatus(
...
+Status GetModelStatusImpl::getAllModelsStatuses(std::map<std::string, tensorflow::serving::GetModelStatusResponse>& modelsStatuses, ModelManager& manager){
+    const std::map<std::string, std::shared_ptr<Model>>& models = manager.getModels();
+    std::map<std::string, tensorflow::serving::GetModelStatusResponse> modelsStatusesTmp;
+    for(auto const& model : models)
+    {
+        std::optional<int64_t> noValueModelVersion;
+        tensorflow::serving::GetModelStatusRequest request;
+        GetModelStatusImpl::createGrpcRequest(model.first, noValueModelVersion, &request);
+        tensorflow::serving::GetModelStatusResponse response;
+        auto status = GetModelStatusImpl::getModelStatus(&request, &response, manager);
```
Comment (rasapala) [inline src/model_service.cpp:144]: Check status here: break and exit on error ?

File: src/test/model_service_test.cpp [hunk 9]
```diff
@@ -313,3 +313,224 @@ TEST(RestModelStatus, serialize2Json) {
     ASSERT_EQ(error_status, StatusCode::OK);
     EXPECT_EQ(json_output, expected_json);
 }
+
+const ovms::ModelConfig DUMMY_MODEL_WITH_ONLY_NAME_CONFIG{
+    "dummy",
+};
+
+TEST(ModelService, getAllModelsStatuses_one_model_one_version) {
```
Comment (rasapala) [inline src/test/model_service_test.cpp:321]: Please add 2 negative tests for:
static Status getAllModelsStatuses(std::map<std::string, tensorflow::serving::GetModelStatusResponse>& models_versions, ModelManager& manager);
    static Status serializeModelsStatuses2Json(const std::map<std::string, tensorflow::serving::GetModelStatusResponse>& models_versions, std::string& output);

**Review by rasapala — APPROVED** [review]
LGTM

**Pull request merged**